core/src/main/java/org/jruby/util/Dir.java

```diff
@@ -776,50 +810,39 @@ private static int addToResultIfExists(Ruby runtime, String cwd, byte[] bytes, i
         FileResource file = JRubyFile.createResource(runtime, cwd, fileName);
 
         if (file.exists()) {
-            byte[] newBytes;
-            
-            // get the real filename (case-sensitive)
-            if(file.isFile() && cwd != null) {
-                String path = file.isSymLink() ? file.absolutePath() : file.canonicalPath();
-
-                // compensate for missing slash
-                if (fileName.endsWith("/")) {
-                    path += "/";
-                }
-
-                if(fileName.contains("./")) {
-                    newBytes = ("./" + path.substring((path.length() - end + 2))).getBytes();
-                } else {
-                    int tempBegin = path.length() - fileName.length();
-                    newBytes = path.substring(tempBegin).getBytes();
-                }
-
-                end = newBytes.length;
-            } else {
-                newBytes = bytes;
-            }
-
-            return func.call(newBytes, begin, end - begin, enc, arg);
+            return func.call(bytes, begin, end - begin, enc, arg);
         }
 
         return 0;
     }
 
     private static int glob_helper(Ruby runtime, String cwd, ByteList path, int sub, int flags, GlobFunc<GlobArgs> func, GlobArgs arg) {
+        int begin = path.getBegin();
+        final int end = begin + path.length();
+        final Encoding enc = path.getEncoding();
+        final byte[] bytes = path.getUnsafeBytes();
+        final byte[] scheme = extractScheme(bytes, begin, end);
+        if (scheme != null) begin += scheme.length;
+        return glob_helper(runtime, cwd, scheme, bytes, begin, end, enc, sub, flags, func, arg);
+    }
+
+    private static int glob_helper(Ruby runtime, String cwd, byte[] scheme, ByteList path, int sub, int flags, GlobFunc<GlobArgs> func, GlobArgs arg) {
         final int begin = path.getBegin();
-        final int end = begin + path.getRealSize();
+        final int end = begin + path.length();
```

Comment:
i'm not sure if this changes anything. looks like length just returns realSize :/

Reply:
@danini-the-panini length() is there to satisfy Java CharSequence.  It is fine to leave it since it will always match byte length but it is a weird method since it implies it should return char length.